File: Assignment42/program5.c
//////////////////////////////////////////////////////////////////////
// File name :       program5.c
// Discription :     It is Recusive program to accept number from user 
//                   and return product of its digits.
// Author :          Mangesh Ashok Bedre
// Date :            17/11/2025
//
//////////////////////////////////////////////////////////////////////

#include<stdio.h>

//////////////////////////////////////////////////////////////////////
// Function Name :   Product
// Discription :     It is used to return product of its digits.
// Input  :          int
// Output :          int
// Author :          Mangesh Ashok Bedre
// Date :            17/11/2025
//
//////////////////////////////////////////////////////////////////////

int Product(int iNo)
{
    static int iMul = 1;
    int iDig = 0;
    
    //Filter
    if(iNo < 0) 
    {
        iNo = - iNo;
    }

    if(iNo == 1)
    {
        return iMul;
    }
    else    
    {
        iDig = iNo % 10;
        iMul = iMul * iDig;
        iNo = iNo / 10;
        Product(iNo);
    }
    
        
}

int main()
{
    int iValue = 0,iRet = 0;

    printf("Enter the number : \n");
    scanf("%d",&iValue);

    iRet = Product(iValue);
    printf("product of digits of %d is %d\n",iValue,iRet);
    return 0;
}

File: Assignment47/program5.java
import java.util.Scanner;
//////////////////////////////////////////////////////////////////////
// File name :       program4.java
// Discription :     It is a program to print  pattern.
// Author :          Mangesh Ashok Bedre
// Date :            25/11/2025
//
//////////////////////////////////////////////////////////////////////
class Pattern
{
    
    //////////////////////////////////////////////////////////////////////
    // Function Name :   Display
    // Discription :     It is used to print pattern.
    // Input  :          int ,int
    // Output :          1 1 1 1
    //                   2 2 2 2
    //                   3 3 3 3
    //                   4 4 4 4
    // Author :          Mangesh Ashok Bedre
    // Date :            25/11/2025
    //
    //////////////////////////////////////////////////////////////////////
    public void Display(int rows, int cols)
    {
        int i = 0;
        int j = 0;

        for(i = 1; i <= rows; i++)
        {
            for(j = 0; j < cols; j++)
            {
                System.out.print(i+"\t");
            }
            System.out.println();
        }
    }
}

class program5
{
    public static void main(String a[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the rows : ");
        int rows = sobj.nextInt();

        System.out.println("Enter the cols : ");
        int cols = sobj.nextInt();

        Pattern pobj = new Pattern();
        pobj.Display(rows,cols);
    }
}